python/sidecar/ai/chat.py

```diff
@@ -22,12 +23,15 @@ def ask_question(
     n_choices = request.n_choices
     temperature = request.temperature
 
-    if user_settings is not None:
-        openai.api_key = user_settings.openai_api_key
-        model = user_settings.openai_chat_model
-    else:
-        openai.api_key = os.environ.get("OPENAI_API_KEY")
+    if user_settings is None:
+        # this is for local dev environment
+        openai.api_key = os.environ.get("API_KEY")
         model = "gpt-3.5-turbo"
+    elif user_settings.model_provider == ModelProvider.OPENAI:
+        openai.api_key = user_settings.api_key
```

Comment:
I think this should be `os.environ["OPENAI_API_KEY"] = user_settings.api_key`: openai.api_key doesn't seem to be used, which is why I was getting a response even when my key was invalid

Reply:
After more investigation, it appears that when `openai.api_key` is invalid, liteLLM falls back to the `OPENAI_API_KEY` env var. I have no clue why my env var is correct (I don't have an .env file) but this is what was happening.
This behavior is a bit confusing, I think we should not fallback to anything if there is an api key set in the app. Carlos told me that the plan was to send the api_key in the request, instead of using `openai.api_key`, maybe this would prevent the fallback. In any case, this is a very edge case because the user has to have an `OPENAI_API_KEY` env var for the fallback to happen, so I don't think it should block this PR, but we have to keep it in mind